.github/workflows/scripts/create-cluster.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+#
+# Creates a cluster on infra.
+#
+
+set -uo pipefail
+
+FLAVOR="$1"
+NAME="$2"
+LIFESPAN="$3"
+WAIT="$4"
+ARGS="$5"
+
+check_not_empty \
+    FLAVOR NAME LIFESPAN WAIT \
+    INFRA_TOKEN
+
+CNAME="${NAME//./-}"
+
+function cluster_info() {
+    infractl 2>/dev/null get "$1" --json
+}
+
+function cluster_status() {
+    cluster_info "$1" | jq -r '.Status'
+}
+
+function cluster_creating() {
+    [ "$(cluster_status "$1")" -eq 1 ]
+}
+
+function infra_status_summary() {
+    gh_summary <<EOF
+*$2*
+Infra status for '$1':
+\`\`\`$(cluster_info "$1")\`\`\`
+
+EOF
+}
+
+case $(cluster_status "$CNAME") in
+1)
+    # Don't wait for the cluster being created, as another workflow could be
+    # waiting for it.
+    # TODO: use concurrency tweak to allow only single workflow running at once.
+    infra_status_summary "$CNAME" "Cluster is being created by another workflow"
+    exit 0
+    ;;
+2)
+    # Cluster exists already.
+    infra_status_summary "$CNAME" "Cluster already exists"
+    exit 0
+    ;;
+4)
+    # Cluster has already been destroyed. Create it again.
+    gh_log notice "Cluster \`$CNAME\` has been destroyed already."
+    infra_status_summary "$CNAME" "Cluster has been destroyed already"
+    ;;
+*)
```

Comment:
I didn't know that when I wrote this, but status `3` means `destroying` - not sure if it is useful here

Reply:
I'll add it to the case for general awareness.

Reply:
Oh. We could wait for it to be fully destroyed and then create it again.

Reply:
> Oh. We could wait for it to be fully destroyed and then create it again.

Yes, in some cases it may be useful to wait instead of failing the workflow.